packages/js-sdk/src/batchCall.js

```diff
@@ -0,0 +1,239 @@
+    getMissingArgumentError,
+    getBatchCallHelpText,
+} = require("./utils/error");
+const { AbiCoder } = require("@ethersproject/abi");
+
+const abiCoder = new AbiCoder();
+
+const OPERATION_TYPES = {
+    ERC20_APPROVE: 1,
+    ERC20_TRANSFER_FROM: 2,
+    SUPERTOKEN_UPGRADE: 101,
+    SUPERTOKEN_DOWNGRADE: 102,
+    SUPERFLUID_CALL_AGREEMENT: 201,
+    CALL_APP_ACTION: 202,
+};
+
+const AGREEMENT_TYPES = {
+    CFA: "cfa",
+    IDA: "ida",
+};
+
+const parseERC20Operation = ({ index, operationType, data }) => {
+    const { token, spender, sender, recipient, amount } = data;
+    if (!amount)
+        throw new Error(
+            getMissingArgumentError("amount", getBatchCallHelpText(index))
+        );
+    if (!token)
+        throw new Error(
+            getMissingArgumentError("token", getBatchCallHelpText(index))
+        );
+
+    /**
+     * @dev ERC20.approve batch operation type
+     * Call spec:
+     * ISuperToken(target).operationApprove(
+     *     abi.decode(data, (address spender, uint256 amount))
+     * )
+     */
+    if (operationType === OPERATION_TYPES.ERC20_APPROVE) {
+        if (!spender)
+            throw new Error(
+                getMissingArgumentError("spender", getBatchCallHelpText(index))
+            );
+        return [
+            operationType,
+            token,
+            abiCoder.encode(["address", "uint256"], [spender, amount]),
+        ];
+    }
+    /**
+     * @dev ERC20.transferFrom batch operation type
+     * Call spec:
+     * ISuperToken(target).operationTransferFrom(
+     *     abi.decode(data, (address sender, address recipient, uint256 amount)
+     * )
+     */
+    if (!sender)
+        throw new Error(
+            getMissingArgumentError("sender", getBatchCallHelpText(index))
+        );
+    return [
+        operationType,
+        token,
+        abiCoder.encode(
+            ["address", "address", "uint256"],
+            [sender, recipient, amount]
+        ),
+    ];
+};
+
+const parseSuperTokenOperation = ({ index, operationType, data }) => {
+    const { amount, token } = data;
+    if (!amount)
+        throw new Error(
+            getMissingArgumentError("amount", getBatchCallHelpText(index))
+        );
+    if (!token)
+        throw new Error(
+            getMissingArgumentError("token", getBatchCallHelpText(index))
+        );
+
+    /**
+     * @dev SuperToken.upgrade batch operation type
+     * Call spec:
+     * ISuperToken(target).operationUpgrade(
+     *     abi.decode(data, (uint256 amount)
+     * )
+     */
+    /**
+     * @dev SuperToken.downgrade batch operation type
+     * Call spec:
+     * ISuperToken(target).operationDowngrade(
+     *     abi.decode(data, (uint256 amount)
+     * )
+     */
+    return [operationType, token, abiCoder.encode(["uint256"], [amount])];
+};
+
+const parseSuperFluidOperation = ({ index, operationType, data }) => {
+    const {
+        superApp,
+        agreementType,
+        method,
+        arguments: args,
```

Comment:
@ngmachado to avoid using reserved keyword "arguments", this is a way to assign a different name to it. See

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment
"Assigning to new variable names"